Tasks/Gulp/gulptask.ts

```diff
@@ -2,118 +2,152 @@
-    if (publishJUnitResults) {
-        //check for pattern in testResultsFiles
-        if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
-            tl.debug('Pattern found in testResultsFiles parameter');
-            var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
-            var allFiles = tl.find(buildFolder);
-            var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
-        }
-        else {
-            tl.debug('No pattern found in testResultsFiles parameter');
-            var matchingTestResultsFiles = [testResultsFiles];
-        }
-        if (!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
-            tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');
-            return 0;
-        }
-        var tp = new tl.TestPublisher("JUnit");
+	if (publishJUnitResults) {
+		//check for pattern in testResultsFiles
+		if (/[\*?+@!{}\[\]]/.test(testResultsFiles)) {
+			tl.debug('Pattern found in testResultsFiles parameter');
+			var matchingTestResultsFiles = tl.findMatch(tl.getVariable('System.DefaultWorkingDirectory'), testResultsFiles, null, { matchBase: true });
+		}
+		else {
+			tl.debug('No pattern found in testResultsFiles parameter');
+			var matchingTestResultsFiles = [testResultsFiles];
+		}
+		if (!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
+			tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');
+			return 0;
+		}
+		var tp = new tl.TestPublisher("JUnit");
 		try {
 			tp.publish(matchingTestResultsFiles, true, "", "", "", true);
 		} catch (error) {
 			tl.warning(error);
 		}
-    }
+	}
 }
 
-function publishCodeCoverage(summaryFile) {
+function publishCodeCoverage(summaryFile, reportDirectory) {
 	try {
 		var ccPublisher = new tl.CodeCoveragePublisher();
 		ccPublisher.publish('cobertura', summaryFile, reportDirectory, "");
 	} catch (error) {
 		tl.warning(error);
 		throw error;
 	}
-}
\ No newline at end of file
+}
+
+function getGulpFiles(gulpFilePath: string): string[] {
+	var gulpFiles = [];
+
+	if (/[\*?+@!{}\[\]]/.test(gulpFilePath)) {
+		gulpFiles = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), gulpFilePath);
+		tl.debug("number of files matching filePath: " + gulpFiles.length);
+		if(gulpFiles.length === 0) {
+			tl.warning(tl.loc('NoGulpFileFound', gulpFilePath));
+		}
+	}
+	else {
+		tl.checkPath(gulpFilePath, "gulpfile");
+		gulpFiles.push(gulpFilePath);
+	}
+	return gulpFiles;
+}
+
+function substituteGulpFilePath(gt: trm.ToolRunner, filePath: string) {
+	var indexOfGulpFile = gt.args.indexOf("--gulpfile");
+	if (indexOfGulpFile >= 0) {
+		gt.args[indexOfGulpFile + 1] = filePath;
+	}
+}
+
+executeTask().catch((error) => tl.setResult(tl.TaskResult.Failed, error));
```

Comment:
I think async/await is the pattern we are moving to in our tasks. So this would become:
```typescript
try {
    await executeTask();
}
catch (err) {
    // ...
}